astropy/table/table.py

```diff
@@ -2132,6 +2132,43 @@ def remove_rows(self, row_specifier):
         if hasattr(self, '_groups'):
             del self._groups
 
+    def iterrows(self, col_names=None):
+        """
+        Return zipped columns for row-wise iteration
+
+        Parameters
+        ----------
+        col_names : list
+        List of column names
+
+         Examples
+        --------
+        Create a table with three columns 'a', 'b' and 'c'::
+
+            >>> t = Table([[1, 2, 3], [0.1, 0.2, 0.3], ['x', 'y', 'z']],
+            ...           names=('a', 'b', 'c'))
+
+        To iterate row-wise using column names
+            >>> for a,b in t.iterrows(['a', 'b'])
+            ...     print(a,b)
+            1 0.1
+            2 0.2
+            3 0.3
+
+        """
+        if col_names is None:
+            raise ValueError('No input Provided')
+
+        for x in col_names:
+            if x not in self.colnames:
+                raise ValueError('Enter Valid Column Names')
```

Comment:
Try to provide useful information in the exception message, e.g. `f"'{name}' is not a valid column name"`.